src/components/Home.js

```diff
@@ -1,20 +1,70 @@
-import React from 'react';
-import { NavLink } from 'react-router-dom';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useCollectionData } from 'react-firebase-hooks/firestore';
+import firebase from '../lib/firebase';
 import getToken from '../lib/tokens';
+import '../styles/Home.css';
+
+const db = firebase.firestore().collection('shopping_list');
 
 const Home = () => {
+  const [existingToken, setExistingToken] = useState('');
+  const history = useHistory();
+
   const newList = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+    history.push('/list');
+  };
+
+  const tokenHandler = (event) => {
+    setExistingToken(event.target.value);
   };
+
+  const [shoppingList] = useCollectionData(
+    db.where('token', '==', existingToken),
+  );
+
+  const submitToken = (e) => {
+    e.preventDefault();
+
+    if (existingToken === '') {
+      alert('Please enter a token...');
+      return;
+    }
+
+    if (shoppingList.length) {
+      localStorage.setItem('token', existingToken);
+      history.push('/list');
+    } else {
+      alert('Token does not exist! Please try again or create a new list.');
+    }
+  };
+
   return (
     <div>
-      <h1>Welcome to Smart Shopping App</h1>
-      <NavLink to="/list">
-        <button type="submit" onClick={newList}>
-          Create List
-        </button>
-      </NavLink>
+      <div className="home">
```

Comment:
I think there might be one too many div wrappers here? Could the outer div be removed and simply the div className="home" be returned?

Reply:
The outer div cannot be removed, all return items have to be inside one tag element e.g. div in this case. I tried to use the className on the outer div but this didn't seem to work. Since it is working in the app.js that the outer div has a className, I will try again.
For the other div wrappers - as for my understanding when tags belong together it is better to surround them in a div tag, for better usage of e.g. flexbox.

Reply:
Ah great!!